src/test/kotlin/com/atkinsondev/opentelemetry/build/OpenTelemetryBuildPluginSpansIntegrationTest.kt

```diff
@@ -0,0 +1,188 @@
+package com.atkinsondev.opentelemetry.build
+
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.jupiter.api.Assertions.assertLinesMatch
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+import org.testcontainers.containers.GenericContainer
+import org.testcontainers.containers.wait.strategy.Wait
+import strikt.api.expectThat
+import strikt.assertions.*
+import java.io.File
+import java.nio.file.Path
+
+class OpenTelemetryBuildPluginSpansIntegrationTest {
+    private val json =
+        Json {
+            ignoreUnknownKeys = true
+        }
+
+    @Serializable
+    class JaegerApiResponse(
+        // In this test this list always contains a single element
+        // Maybe this is a list when querying for multiple traces?
+        val data: List<JaegerApiResponseTrace>,
+    )
+
+    @Serializable
+    class JaegerApiResponseTrace(
+        // Flat list of all spans in the trace
+        // Order isn't necessarily correct
+        val spans: List<JaegerApiResponseSpan>,
+    )
+
+    @Serializable
+    class JaegerApiResponseSpan(
+        val spanID: String,
+        val operationName: String,
+        val references: List<JaegerApiResponseSpanReference>,
+        val startTime: Long,
+    )
+
+    @Serializable
+    class JaegerApiResponseSpanReference(
+        // Not used currently, but we currently only handle CHILD_OF correctly.
+        // Decode into enum only containing CHILD_OF to get an exception when for some reason another relationship type is returned in the future.
+        @Suppress("unused")
+        val refType: RefType,
+        val spanID: String,
+    ) {
+        @Suppress("unused") // Future
+        @Serializable
+        enum class RefType {
+            CHILD_OF,
+        }
+    }
+
+    class FlatSpan(
+        val operationName: String,
+        val startTime: Long,
+        val depth: () -> Int,
+    )
+
+    companion object {
+        private const val OTLP_GRPC_PORT = 4317
+        private const val JAEGER_HEALTHCHECK_PORT = 14269
+        private const val JAEGER_QUERY_PORT = 16686
+    }
+
```

Comment:
When this approach to testing may be used in multiple tests in the future, those `@Serializable` classes should probably moved to their own file(s).